src/wallet/rpcwallet.cpp

```diff
@@ -387,23 +387,22 @@ UniValue listaddresses(const UniValue& params, bool fHelp)
     // This will include any address that has been generated by this wallet.
     for (const std::pair<CTxDestination, CAddressBookData>& item : pwalletMain->mapAddressBook) {
         t_generated_dests.insert(item.first);
+        auto script = GetScriptForDestination(item.first);
+        if (pwalletMain->HaveWatchOnly(script)) {
+            t_watchonly_dests.insert(item.first);
+        }
```

Comment:
This is going to show the address as both generated and watch-only. That doesn't seem correct; the address should be in one or the other, but not both.